chore(blog): allow building with standard hugo

nikku opened the pull request:
* workaround #37 by checking the hugo version before including
  RSS feed. This should not impact generation on CI (?), as we
  use our custom, monkeypatched hugo there?
* add grunt-cli as dependency so that `npm run dev` just works for all users.

__NOTE:__ This should be reviewed by someone using Windows with the custom hugo configured to see if bundling still works as expected.

AndreaGiardini commented on theme-src/layouts/partials/header.html:
```diff
@@ -14,7 +14,9 @@
 
+  {{ if le .Hugo.Version "0.17.0" }}
   {{ if .RSSlink }}<link href="{{ .RSSlink }}" rel="alternate" type="application/rss+xml" title="{{ .Title }}" />{{ end }}
+  {{ end }}
```
I would add an `else` here. You can use the same line of template replacing `RSSlink` with `RSSLink`

nikku commented:
Did not know about that :1st_place_medal:

nikku commented:
Fixed.

sdorokhova commented:
Hi @nikku,
in README it's written that we must compile (`npm run build`) `theme-src` directory, so that the changes are applied to `themes` directory. And then push all the changes. 
I pushed `header.html` in 01de448. 
Standard Hugo v0.30.2 windows/amd64 works fine for me.

BUT after compiling I also got these files changed:
- bpmn-viewer.js
- scripts.js

Is it because of new dependency you added? I don't feel confident of committing these changes, as I know nothing about JavaScript. Could you please finalize it? You would need to compile the files and push the changes, if everything is fine.

Thank you

AndreaGiardini commented:
We also need @hawky-4s- to change the version of hugo that we use in Jenkins for the compilation before merging this

camunda-jenkins commented:
Which version do we need?

________________________________
Von: Andrea Giardini <notifications@github.com>
Gesendet: Donnerstag, 9. November 2017 08:39:42
An: camunda/blog.camunda.org
Cc: Subscribed
Betreff: Re: [camunda/blog.camunda.org] chore(blog): allow building with standard hugo (#41)


We also need @hawky-4s-<https://github.com/hawky-4s-> to change the version of hugo that we use in Jenkins for the compilation before merging this

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/camunda/blog.camunda.org/pull/41#issuecomment-343071751>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHI8YRC_41OC7vXIpsvgKBvqhhQMHgcMks5s0qw-gaJpZM4QIoO6>.

hawky-4s- commented:
@nikku: If we don't use the partials feature of the monkeypatched camunda hugo version, then updating the hugo version is no problem.

nikku commented:
@sdorokhova The dependency I've added has nothing to do with the update script files.

Locally rebuilding doesn't change these files, too:

```bash
~/camunda/projects/blog.camunda.org modern-hugo-build*
❯ rm -rf node_modules

~/camunda/projects/blog.camunda.org modern-hugo-build*
❯ npm i
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN grunt-browserify@2.1.0 requires a peer of browserify@^4.0.0 but none is installed. You must install peer dependencies yourself.

added 816 packages in 11.09s

~/camunda/projects/blog.camunda.org modern-hugo-build* 11s
❯ npm run build
...
Done, without errors.
Started building sites ...

Built site for language en:
0 draft content
0 future content
0 expired content
232 regular pages created
126 other pages created
3 non-page files copied
207 paginator pages created
53 tags created
6 categories created
total in 1692 ms

~/camunda/projects/blog.camunda.org modern-hugo-build* 6s
❯ git status
On branch modern-hugo-build
Your branch is up to date with 'origin/modern-hugo-build'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        package-lock.json

nothing added to commit but untracked files present (use "git add" to track)
```

The `package-lock.json` can be ignored, it is generated because I `npm@5`.

Mind that I clean the `node_modules` folder and perform a fresh `npm install` to verify this.

sdorokhova commented:
Hi @nikku ,
thanks for checking. I've merged the changes within 55381cd3ccb333e100db842092876299bf2317cb .
Blog publishing went well. 
@AndreaGiardini @hawky-4s- , if you have any objections to me pushing this, please tell.
